juriscraper/pacer/utils.py

```diff
@@ -59,9 +60,18 @@ def get_pacer_case_id_from_doc1_url(url):
         return None
 
 
-def get_pacer_magic_num_from_doc1_url(url):
-    """Extract the caseid from the doc1 URL."""
-    match = re.search(r"magic_num=(\d+)", url)
+def get_pacer_magic_num_from_doc1_url(
+    url: str, email_notice_type: Optional[str] = None
+) -> Optional[str]:
+    """Extract the magic number from the doc1 URL."""
+    if email_notice_type == "NDA":
+        # NDA free look link format is:
+        # https://ecf.ca2.uscourts.gov/docs1/00208721516?uid=b775e9908ad79ce2
```

Comment:
Huh. I'm not familiar with the NDAs enough. You feel good about this being the magic number?

Reply:
Yes, at least within the NDAs examples we have now if there is a document link available it has this format where `uid=b775e9908ad79ce2` is the "magic number".

In fact, if you try to go to this link (anonymously) that it's already used you'll see the following:
[https://ecf.ca2.uscourts.gov/docs1/00208721516?uid=b775e9908ad79ce2](https://ecf.ca2.uscourts.gov/docs1/00208721516?uid=b775e9908ad79ce2)
![Screen Shot 2022-08-12 at 18 09 41](https://user-images.githubusercontent.com/486004/184455511-046f6902-ed00-4ac3-b3f4-ac72fec3f1d2.png)